### PR TITLE
refactor(patients): use async/await

### PR DESCRIPTION
**Pull request by jniles**
This commit migrates the patients controller on the server to use async/await where possible.  It also updates the patient reference calculation to pull the values from the entity_map table.  Finally, it makes sure that the entity_map table is kept up to date with any changes made to the patient name.

Contributes to #2458.

### CODE REVIEW

**Comment by jniles** [comment]
@jeremielodi, can I get a review?

File: server/controllers/medical/patients/index.js [hunk 1]
```diff
@@ -229,22 +223,19 @@ function lookupPatient(patientUuid) {
...
-      return patient;
-    });
+  _.extend(patient, {
+    barcode : barcode.generate(identifiers.PATIENT.key, patient.uuid),
+  });
+
+  const priceList = await lookupPatientPriceList(buid);
+  patient.price_list_uuid = patient.price_list_uuid || priceList;
+  return patient;
 }
```
Comment (jeremielodi) [inline server/controllers/medical/patients/index.js:240]: This is as clear as a day

File: server/controllers/medical/patients/index.js [hunk 2]
```diff
@@ -462,9 +452,9 @@ function find(options) {
   filters.equals('uuid');
 
   // filters for location
-  const orignSql = `(originVillage.name LIKE ?) OR (originSector.name LIKE ?) OR (originProvince.name LIKE ?)`;
-  const params1 = _.fill(Array(3), `%${options.originLocationLabel || ''}%`);
-  filters.custom('originLocationLabel', orignSql, params1);
+  const originNameSql = `(originVillage.name LIKE ?) OR (originSector.name LIKE ?) OR (originProvince.name LIKE ?)`;
+  const originNameParams = _.fill(Array(3), `%${options.originLocationLabel || ''}%`);
```
Comment (jeremielodi) [inline server/controllers/medical/patients/index.js:456]: 👍

**Review by jeremielodi — COMMENTED** [review]
This PR looks great, using async/await makes the code clear and more readable. Good job @jniles!

**Pull request merged**